REDCapUITweaker.php

```diff
@@ -1963,6 +2196,63 @@ function provideUserRoleName()
 
 
 
+	// Output JavaScript to provide versionless URLs.
+
+	function provideVersionlessURLs()
+	{
+
+?>
+<script type="text/javascript">
+  $(function()
+  {
+    var vVersionIndex = window.location.href.indexOf( 'redcap_v' + redcap_version )
+    if ( vVersionIndex != -1 )
+    {
+      var vFullURL = window.location.href
+      $('form').each( function()
+      {
+        if ( $(this).attr('action') === undefined )
+        {
+          if ( typeof this.action == 'string' )
+          {
+            this.action = this.action // set implicit action explicitly
+          }
+          else
+          {
+            this.action = vFullURL.replace(/#.*$/,'')
+          }
+        }
+      })
+      var vBaseElem = $('<base>')
+      vBaseElem.attr('href',vFullURL)
+      $('head').append( vBaseElem )
+      var vOldURL = vFullURL.slice( 0, vVersionIndex + 8 + redcap_version.length )
```

Comment:
I know Nick does not like these hard coded values, can you add as constant?

Reply:
I generally don't like string literals, but in this case I'm not too worried about replacing 'redcap' with a constant, as this is so baked into the redcap system.